### PR TITLE
refactor(batch): Remove unnecessary executor

### PR DESCRIPTION
**Pull request by liurenjie1024**
## What's changed and what's your intention?

Remove unnecessary executor after removing legacy java client.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

File: src/compute/tests/table_v2_materialize.rs [hunk 1]
```diff
@@ -117,17 +117,17 @@ async fn test_table_v2_materialize() -> Result<()> {
...
-        source_manager.clone(),
-        table_columns,
-        "CreateTableExecutor".to_string(),
-        Info::TableSource(Default::default()),
-    );
-    // Execute
-    create_table.open().await?;
-    create_table.next().await?;
-    create_table.close().await?;
+    // let mut create_table = CreateTableExecutor::new(
```
Comment (neverchanje) [inline src/compute/tests/table_v2_materialize.rs:120]: Just delete them if not used.

**Comment by liurenjie1024** [comment]
After this PR merged, we've got to restore several test:
1. test_table_v2_materialize cc @BugenZhao 
2. task/test_util.rs cc @neverchanje

File: src/compute/tests/table_v2_materialize.rs [hunk 2]
```diff
@@ -85,7 +84,8 @@ impl BatchExecutor for SingleChunkExecutor {
 
 /// This test checks whether batch task and streaming task work together for `TableV2` creation,
 /// insertion, deletion, and materialization.
-#[tokio::test]
+// #[tokio::test]
+#[allow(dead_code)]
```
Comment (BugenZhao) [inline src/compute/tests/table_v2_materialize.rs:88]:
```suggestion
#[ignore]
#[tokio::test]
```

**Review by BugenZhao — APPROVED** [review]
Cool!

File: src/batch/src/task/test_utils.rs [hunk 3]
```diff
@@ -192,46 +192,46 @@ impl<'a> TableBuilder<'a> {
...
-                identity: "CreateTableExecutor".to_string(),
-            }),
-
-            exchange_info: Some(ExchangeInfo {
-                mode: 0,
-                distribution: None,
-            }),
-        }
-    }
+    // fn build_create_table_plan(&self) -> PlanFragment {
```
Comment (liurenjie1024) [inline src/batch/src/task/test_utils.rs:206]: Is this file still usable? @neverchanje

**Review by BowenXiao1999 — APPROVED** [review]

File: src/compute/tests/table_v2_materialize.rs (hunk 1 above)
Comment (yezizp2012) [reply src/compute/tests/table_v2_materialize.rs]: FYI, I fixed this test in this PR: https://github.com/singularity-data/risingwave/pull/2550

File: src/batch/src/task/test_utils.rs (hunk 3 above)
Comment (neverchanje) [reply src/batch/src/task/test_utils.rs:206]: I've confirmed that TestRunner is completely unused. Pls remove them.

**Pull request merged**